lib/site.post-type.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * SitePostType class
+ */
+export default class SitePostType {
+	/**
+	 * Create a SitePostType instance
+	 *
+	 * @param {String} postType - post type
+	 * @param {String} siteId - site id
+	 * @param {WPCOM} wpcom - wpcom instance
+	 * @return {Null} null
+	 */
+	constructor( postType, siteId, wpcom ) {
+		if ( ! siteId ) {
+			throw new TypeError( '`siteId` is not correctly defined' );
+		}
+
+		if ( ! postType ) {
+			throw new TypeError( '`postType` is not correctly defined' );
+		}
+
+		if ( ! ( this instanceof SitePostType ) ) {
+			return new SitePostType( postType, siteId, wpcom );
+		}
+
+		this.wpcom = wpcom;
+
+		this._siteId = encodeURIComponent( siteId );
+		this._postType = encodeURIComponent( postType );
+		this._rootPath = `/sites/${ this._siteId }/post-types/${ this._postType }`;
+	}
+
+	/**
+	 * Get a list of taxonomies for the post type
+	 *
+	 * @param {Function} fn - callback function
+	 * @return {Promise} Promise
+	 */
+	taxonomiesList( fn ) {
```

Comment:
Are we pretty sure that `query` object here isn't needed? What if we wanna filter by `fields`for instance?

```es6
wpcom
  .site( 'es.blog.wordpress.com' )
  .postType( 'post' )
  .taxonomiesList( { fields: 'name. public' } ).then( list => // ... )
```

Add `query` parameter is pretty simple, and this is optional when the method is invoked.

```es6
taxonomiesList( query, fn ) {
  const termsPath = `${ this._rootPath }/taxonomies`;
  return this.wpcom.req.get( termsPath, query, fn );
}